src/dispatch-payload.ts

```diff
@@ -30,13 +27,25 @@ type GetFunctionFileCallback = {
   (functionCallbackId: string): string | FunctionModule;
 };
 
+// A helper structure that maps event names to the functions responsible for invoking userland code with the relevant event's payload
+const EVENT_TO_HANDLER_MAP = {
+  [EventTypes.FUNCTION_EXECUTED]: RunFunction,
+  [EventTypes.BLOCK_ACTIONS]: RunBlockAction,
+  [EventTypes.BLOCK_SUGGESTION]: RunBlockSuggestion,
+  [EventTypes.VIEW_CLOSED]: RunViewClosed,
+  [EventTypes.VIEW_SUBMISSION]: RunViewSubmission,
+};
+
 export const DispatchPayload = async (
   // deno-lint-ignore no-explicit-any
   payload: InvocationPayload<any>,
   hookCLI: Protocol,
   getFunctionFile: GetFunctionFileCallback,
 ) => {
-  const eventType = payload?.body?.event?.type || payload?.body?.type || "";
+  // TODO: should we check that this is a ValidEventType at runtime?
```

Comment:
Unclear if this TODO is valuable or not. IIRC there may be a reason we don't put expectations on the event type, in order to allow for prototyping random new event handling in the runtime?